spec/v2/providers/firestore.spec.ts

```diff
@@ -511,6 +513,194 @@ describe("firestore", () => {
     });
   });
 
+  describe("onDocumentWrittenWithAuthContext", () => {
```

Comment:
Missing `it calls init function` cases everywhere. Because v2 doesn't have a shared `makeCloudFunction` library, I have tests for each event type.